klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+
+    ## list_res_db_files.append(run_check(drc_path, "antenna", layout_path, drc_run_dir, switches))
+    if not arguments["--table"]:
+        list_of_tables = get_list_of_tables(rule_deck_full_path)
+    else:
+        list_of_tables = arguments["--table"]
+
+    ## Generate run rule deck from template.
+    for t in list_of_tables:
+        drc_file = generate_drc_run_template(rule_deck_full_path, drc_run_dir, [t])
+        list_rule_deck_files[t] = drc_file
+
+    ## Run All DRC files.
+    list_res_db_files = []
+    with concurrent.futures.ThreadPoolExecutor(max_workers=os.cpu_count()) as executor:
+        future_to_run_name = dict()
+        for n in list_rule_deck_files:
+            future_to_run_name[
+                executor.submit(
+                    run_check,
+                    list_rule_deck_files[n],
+                    n,
+                    layout_path,
+                    drc_run_dir,
+                    switches,
+                )
+            ] = n
+
+        for future in concurrent.futures.as_completed(future_to_run_name):
+            run_name = future_to_run_name[future]
+            try:
+                list_res_db_files.append(future.result())
+            except Exception as exc:
+                logging.error("%s generated an exception: %s" % (run_name, str(exc)))
+                traceback.print_exc()
+
+    ## Check run
+    check_drc_results(list_res_db_files)
+
+
+def run_single_processor(
+    arguments: dict,
+    rule_deck_full_path: str,
+    layout_path: str,
+    switches: dict,
+    drc_run_dir: str,
+):
+    """
+    run_single_processor run the drc checks as single run.
+
+    Parameters
+    ----------
+    arguments : dict
+        Dictionary that holds the arguments passed to the run_drc script.
+    rule_deck_full_path : str
+        String that holds the path of the rule deck files.
+    layout_path : str
+        Path to the target layout.
+    switches : dict
+        Dictionary that holds all the switches that will be passed to klayout run.
+    drc_run_dir : str
+        Path to the run location.
+    """
+
+    list_res_db_files = []
+
+    ## Run Antenna if required.
+    if arguments["--antenna"] or arguments["--antenna_only"]:
+        drc_path = os.path.join(rule_deck_full_path, "rule_decks", "antenna.drc")
+        list_res_db_files.append(
+            run_check(drc_path, "antenna", layout_path, drc_run_dir, switches)
+        )
+
+        if arguments["--antenna_only"]:
+            logging.info("## Completed running Antenna checks only.")
+            exit()
+
+    ## Run Density if required.
+    if arguments["--density"] or arguments["--density_only"]:
+        drc_path = os.path.join(rule_deck_full_path, "rule_decks", "density.drc")
+        list_res_db_files.append(
+            run_check(drc_path, "density", layout_path, drc_run_dir, switches)
+        )
+
+        if arguments["--density_only"]:
+            logging.info("## Completed running density checks only.")
```

Comment:
should we print the result in `list_res_db_files`?

Reply:
Not sure what you want here. I'm confused.

Reply:
before existing we append something to `list_res_db_files` but we don't make any use of it, is that intentional?

Reply:
@proppy removed.

Reply:
don't see it removed, but I guess that's fine: I just wanted to check if displaying `list_res_db_files` was required here before exiting.